mygrad/math.py

```diff
@@ -115,3 +115,25 @@ def subtract(a, b):
         -------
         mygrad.Tensor"""
     return Tensor._op(Subtract, a, b)
+
```

Comment:
Can you leave the `dot` and `Dot` (operation) changes out of this PR? Let's have it just be of the GRU, and discuss the `dot` operation separately.

My only hang up with it is that people expect `dot` to behave like numpy-dot which covers both vector dot products and matrix multiplication - the latter is covered by the `dense` layer. So I just want to think about how to deal with this design issue before committing.